easybuild/easyconfigs/p/Python/Python-3.11.5-custom-ctypes.patch

```diff
@@ -0,0 +1,52 @@
+diff -ruN Python-3.11.5.orig/Lib/ctypes/__init__.py Python-3.11.5/Lib/ctypes/__init__.py
+--- Python-3.11.5.orig/Lib/ctypes/__init__.py	2025-06-03 17:24:09.817225787 +0200
++++ Python-3.11.5/Lib/ctypes/__init__.py	2025-06-03 17:26:02.851166381 +0200
+@@ -13,6 +13,8 @@
+ from _ctypes import ArgumentError
+ 
+ from struct import calcsize as _calcsize
++from ctypes import util
++import re
+ 
+ if __version__ != _ctypes_version:
+     raise Exception("Version number mismatch", __version__, _ctypes_version)
+@@ -349,6 +351,11 @@
+             flags |= _FUNCFLAG_USE_ERRNO
+         if use_last_error:
+             flags |= _FUNCFLAG_USE_LASTERROR
++        if _os.name == "posix":
++            if name and name.endswith(".so"):
++                s = re.sub(r'lib', '', name)
++                s = re.sub(r'\..*', '', s)
++                self._name=util._findLib_ld(s) 
```

Comment:
I'm not sure I understand what's happening here. What issue is this actually resolving? Also, invoking _findLib_ld means we need to make absolutely sure that _that_ searches the correct path (though I guess that's ensured by the changes below as well)

Reply:
oooh, wait, this is where you overwrite the name with the full path already?

Reply:
Yes, by overriding `self._name`